ldon/README.md

```diff
@@ -0,0 +1,46 @@
+# LDON: Lurk Data Object Notation
+
+The Lurk Data Object Notation, or LDON (pronounced
+"[Ladon](https://en.wikipedia.org/wiki/Ladon_(mythology)"), is a
+content-addressable linked data format designed for zero knowledge contexts,
+such as the Lurk zkSNARK language.
+
+LDON is a simple syntax over Lurk expressions, and is compatible with JSON and
+IPLD.
+
+The primary distinction between LDON and Lurk proper is that LDON makes no
+assumptions about reserved symbols or reduction rules. 
+
+For example, the Lurk expression `(lambda (x) x)` corresponds to the LDON
+expression `(LURK.LAMBDA (X) X)` , but in Lurk this will evaluate to a function
+object, whereas `ldon` considers expressions to be static. Furthermore, `ldon`
+is agnostic to what is built on top of it. One could define a language with
+evaluation rules totally different from Lurk, where functions are built out of
+expressions that look like the `ldon` expression `(fun x => x)`.
+
+LDON syntax is defined as follows:
+
+```rust
+pub enum Syn<F: LurkField> {
+  // A field element: 1, 0xff
+  Num(Pos, F),
+  // A u64 integer: 1u64, 0xffu64
+  U64(Pos, u64),
+  // A hierarchical symbol: foo, foo.bar.baz
+  Symbol(Pos, Vec<String>),
```

Comment:
this is a distinction present in lurk-rs: https://github.com/lurk-lang/lurk-rs/blob/f31bccab84c95c68ccf21b0b27b47f9cec949d79/src/sym.rs#L17 and corresponds to keywords in Common Lisp: https://stackoverflow.com/questions/23969203/what-is-the-difference-between-a-keyword-symbol-and-a-quoted-symbol

It's equivalent to if we had a bool flag on Symbol, but because it has a different tag value (and syntax) I've factored it out separately